src/category_theory/structured_arrow.lean

```diff
@@ -83,6 +80,16 @@ def iso_mk {f f' : structured_arrow S T} (g : f.right ≅ f'.right)
   (w : f.hom ≫ T.map g.hom = f'.hom) : f ≅ f' :=
 comma.iso_mk (eq_to_iso (by ext)) g (by simpa [eq_to_hom_map] using w.symm)
 
+/-- Eta rule for structured arrows. Prefer `structured_arrow.eta`, since equality of objects tends
+    to cause problems. -/
+lemma eq_mk (f : structured_arrow S T) : f = mk f.hom :=
+by { cases f, congr, ext, }
+
+/-- Eta rule for structured arrows. -/
+@[simps]
+def eta (f : structured_arrow S T) : f ≅ mk f.hom :=
```

Comment:
Should this be called `iso_mk`, just like above?

Reply:
Unfortunately, the name `iso_mk` is already taken, by a helper that creates isomorphisms in the category of structured arrows. There are several possibilities:
* Use `iso_mk` for the helper and `eta` for this definition (cf. `prod.eta` etc.)
* Use `iso.mk` for the helper and `iso_mk` for this definition
* Use `mk_iso` for the helper and `iso_mk` for this definition
* Something else

Which of these options do you prefer?

Reply:
I thought that `iso_mk` was typically used for constructing isomorphisms in some category of objects structured in some way by taking an isomorphism of the underlying object together with a proof that it makes sense (eg in over.lean, structured_arrow.lean, endofunctor/algebra.lean, monad/algebra.lean etc); which suggests that the new `iso_mk` for cones should be renamed instead (I quite like `eta` for that).

Reply:
Sounds sensible, I renamed `(co)cones.iso_mk` to `(co)cones.eta`.